client/packages/common/src/ui/layout/tables/material-react-table/useTableLocalStorage.ts

```diff
@@ -41,18 +42,31 @@ export const useTableLocalStorage = <T extends MRT_RowData>(
   const debouncedColumnSizing = useDebouncedValue(columnSizing, 1000);
 
   useEffect(() => {
-    const hidden = Object.entries(columnVisibility)
-      .filter(([_, isVisible]) => !isVisible)
-      .map(([columnId]) => columnId);
+    const savedString = localStorage.getItem(
+      `@openmsupply-client/tables/${tableId}`
+    );
+    const existingCustomisations = savedString ? JSON.parse(savedString) : {};
+    const initial = table.initialState;
 
     localStorage.setItem(
       `@openmsupply-client/tables/${tableId}`,
       JSON.stringify({
         density,
-        hidden,
-        pinned: columnPinning,
         columnSizing: debouncedColumnSizing,
-        columnOrder,
+
+        // These can change with different preferences, or versions of OMS
+        // We should only save them when the user actually customises something
```

Comment:
When nothing in local storage, we have some defaults which populate table `initialState` - in previous version, these defaults were then being saved as the user customisation.

If you then enabled different prefs, we were still respecting that customisation. This logic instead determines that if the table state hasn't changed from whatever was it's initial state, we don't need to save anything (so if the defaults change, user will see that)